ecc/bls12381/g1.go

```diff
@@ -392,7 +392,8 @@ func G1Generator() *G1 {
 }
 
 // affinize converts an entire slice to affine at once
-func affinize(points []*G1) {
+func affinize(points []*G1) (out []G1) {
```

Comment:
What about not declaring `out` as a return variable, but declaring it on the line below? My sense is there is one less allocation, and the code reads the same.
Another question is about returning `[] *G1`, but I understand motivations reason to stick to `[]G1`

Reply:
To avoid allocations you could pass it as an argument.

Reply:
I've tried passing the `out` slice as a parameter, but still one allocation is needed.